File: code_snippets/ts/src/concepts/invocations/rpc.ts
import * as restate from "@restatedev/restate-sdk";
import { greetCounter, greeter, myWorkflow } from "./utils";

// <start_rpc_call>
async function myRestateHandler(ctx: restate.Context) {
  // !focus(1:2)
  // From a Restate handler
  const greet = await ctx.serviceClient(greeter).greet({ greeting: "Hi" });
}
// <end_rpc_call>

File: code_snippets/python/src/concepts/invocations/rpc.py
from restate import Service, Context

from src.concepts.invocations.utils import greet

my_service = Service("MyService")


# <start_rpc_call>
@my_service.handler()
async def my_handler(ctx: Context, arg):
    # !focus
    greeting = await ctx.service_call(greet, arg="Hi")


# <end_rpc_call>
